doc/source/developer_overview.rst

```diff
@@ -143,43 +143,48 @@ The ``ament_lint`` Repository
 
 Because ``ament_tools`` and ``ament_package`` are ament Python packages, they can be bootstrapped in an ament workspace just like any other |package|.
 
-ROS Implementation Interfaces
------------------------------
+Ideally these tools could be combined with the tools provided by `catkin_tools <https://github.com/catkin/catkin_tools>`_ in order to provide a single tool for both build systems.
+We believe this is possible because the two build systems are so similar, but there has not been time to solve the remaining issues and actually consolidate the tools yet.
 
-ROS implementation interfaces are public C |APIs| that are not directly used by the typical ROS user, but instead are used to implement |client libraries|.
-The ROS |client libraries| provide the user facing |APIs| that most ROS users are used to seeing, perhaps in a variety of programming languages.
+Internal ROS Interfaces
+-----------------------
 
-There are two main implementation interfaces: the ROS middleware interface (``rmw`` |API|) and the ROS |client library| interface (``rcl`` |API|).
-The ``rmw`` |API| is the main point at which the ROS 2 software stack touches the middleware implementation.
-The middleware used for ROS 2 is DDS, and a middleware implementation is one of the DDS vendors.
-The ``rcl`` |API| is a slightly higher level |API| which is used to implement the |client libraries| and does not touch the middleware implementation directly.
+The internal ROS interfaces are public C |APIs| that are intended for use by developers who are creating |client libraries| or adding a new underyling middleware, but not inteded for use by typical ROS users.
+The ROS |client libraries| provide the user facing |APIs| that most ROS users are familiar with, and may come in a variety of programming languages.
 
-In addition to these two static |APIs| there is the ``rosidl`` |API|, which is more complicated because it is usually generated for each message and because it can be exposed directly to users.
-The ``rosidl`` *interface* simply defines what code is to be generated for messages in each programming language that has a |client library|, while the ``rosidl`` *implementation* contains the code for generating that code, i.e. converting ROS message types to and from native types for the middleware implementation (vendor specific DDS types).
-The ``rosidl`` C implementation can be reused by other languages that can interface with C, e.g. Python.
+There are two main internal interfaces:
+
+- the ROS middleware interface (``rmw`` |API|)
+- the ROS client library interface (``rcl`` |API|)
+
+The ``rmw`` |API| is the interface between the ROS 2 software stack and the underlying middleware implementation.
+The underlying middleware used for ROS 2 is either a DDS or RTPS implemenation, and is responsible for discovery, publish and subscribe mechanics, service request-reply mechanics, and serialization of message types.
```

Comment:
I'll try to put in an analogy or something to help with this distinction. Maybe "RTPS is to DDS as ROSTCP is to roscpp"?